server/hdf5dtype.py

```diff
@@ -296,28 +296,28 @@ def createBaseDataType(typeItem):
         dtName = getNumpyTypename(typeItem)
         dtRet = np.dtype(dtName)
         return dtRet  # return predefined type
-        
+
     if type(typeItem) != dict:
         raise TypeError("Type Error: invalid type")
-        
-        
+
+
     if 'class' not in typeItem:
         raise KeyError("'class' not provided")
     typeClass = typeItem['class']
-    shape = ''
-    if 'shape' in typeItem:  
-        dims = None      
-        if type(typeItem['shape']) == int:
-            dims = (typeItem['shape'])  # make into a tuple
-        elif type(typeItem['shape']) not in (list, tuple):
+    shape = typeItem.get('shape', typeItem.get('dims', ''))
```

Comment:
H5T_ARRAY's JSON uses ``dims`` key not ``shape``. I tried to avoid complicating much so the ``shape`` key's value is retrieved first, if that does not exist then ``dims`` key's value second, if that does not exist then the default is empty string: ``''``.